parquet-arrow/src/main/java/org/apache/parquet/arrow/schema/SchemaConverter.java

```diff
@@ -381,41 +390,41 @@ public TypeMapping convertINT32(PrimitiveTypeName primitiveTypeName) throws Runt
           return integer(32, true);
         }
         switch (ot) {
-        case INT_8:
-          return integer(8, true);
-        case INT_16:
-          return integer(16, true);
-        case INT_32:
-          return integer(32, true);
-        case UINT_8:
-          return integer(8, false);
-        case UINT_16:
-          return integer(16, false);
-        case UINT_32:
-          return integer(32, false);
-        case DECIMAL:
-          return decimal(type.getDecimalMetadata());
-        case DATE:
-          return field(new ArrowType.Date());
-        case TIMESTAMP_MICROS:
-          return field(new ArrowType.Timestamp(TimeUnit.MICROSECOND));
-        case TIMESTAMP_MILLIS:
-          return field(new ArrowType.Timestamp(TimeUnit.MILLISECOND));
-        case TIME_MILLIS:
-          return field(new ArrowType.Time());
-        default:
-        case TIME_MICROS:
-        case INT_64:
-        case UINT_64:
-        case UTF8:
-        case ENUM:
-        case BSON:
-        case INTERVAL:
-        case JSON:
-        case LIST:
-        case MAP:
-        case MAP_KEY_VALUE:
-          throw new IllegalArgumentException("illegal type " + type);
+          case INT_8:
+            return integer(8, true);
+          case INT_16:
+            return integer(16, true);
+          case INT_32:
+            return integer(32, true);
+          case UINT_8:
+            return integer(8, false);
+          case UINT_16:
+            return integer(16, false);
+          case UINT_32:
+            return integer(32, false);
+          case DECIMAL:
+            return decimal(type.getDecimalMetadata());
+          case DATE:
+            return field(new ArrowType.Date(DateUnit.DAY));
+          case TIMESTAMP_MICROS:
+            return field(new ArrowType.Timestamp(org.apache.arrow.vector.types.TimeUnit.MICROSECOND, "UTC"));
+          case TIMESTAMP_MILLIS:
+            return field(new ArrowType.Timestamp(org.apache.arrow.vector.types.TimeUnit.MILLISECOND, "UTC"));
+          case TIME_MILLIS:
+            return field(new ArrowType.Date(DateUnit.MILLISECOND));
```

Comment:
Why not Time(MILLISECOND) ?

Reply:
@julienledem I made a mistake. I will fix it. Thanks.